tee-launcher/launcher.py

```diff
@@ -141,62 +296,113 @@ def main():
     if proc.returncode:
         raise RuntimeError("docker run non-zero exit code %d", proc.returncode)
 
-def get_manifest_digest(registry_url: str, image_name: str, tags: list[str], image_digest: str):
-    '''Given an `image_digest` returns a manifest digest.
+
+def request_until_success(url: str, headers: Dict[str, str],
+                          rpc_request_interval_secs: float,
+                          rpc_request_timeout_secs: float,
+                          rpc_max_attempts: int) -> Response:
+    """
+    Repeatedly sends a GET request to the specified URL until a successful (200 OK) response is received.
+
+    Args:
+        url (str): The URL to request.
+        headers (Dict[str, str]): Optional headers to include in the request.
+        rpc_request_interval_secs (float): Time in seconds to wait between retries on failure.
+        rpc_request_timeout_secs (float): Maximum time in seconds to wait for a request to succeed.
+
+    Returns:
+        Response: The successful HTTP response object with status code 200.
+
+    Notes:
+        - Retries indefinitely until the request succeeds.
+        - Prints a warning with the response content on each failure.
+    """
+    for attempt in range(1, rpc_max_attempts + 1):
+        # we sleep at the beginning, to ensure that we respect the timeout. Performance is not a priority in this case.
+        time.sleep(rpc_request_interval_secs)
```

Comment:
We could consider increasing this interval on each loop by a factor of e.g. 1.5 until we hit a max interval of 1 minute (or any value that seems sensible).